### PR TITLE
Add missing changelog entries and minor formatting improvements.

### PR DESCRIPTION
**Pull request by ioquatix**
@jeremyevans I added myself to the lower case headers, because this is something I felt like I deeply cared about and pushed for and attempted several times to implement but you managed to do it in the end.

I changed both of us to link references to make it easier to read the markdown. I suggest we do this for other common authors going forward (and maybe PR/Issue links).

I've reformatted some of the code constants with backticks.

There was one missing spec change: response headers' values can be an array.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,60 +6,62 @@ All notable changes to this project will be documented in this file. For info on
...
-- Remove `rack.multithread`/`rack.multiprocess`/`rack.run_once`. These variables generally come too late to be useful. ([#1720](https://github.com/rack/rack/pull/1720), [@ioquatix](https://github.com/ioquatix), [@jeremyevans](https://github.com/jeremyevans)))
-- Remove deprecated Rack::Request::SCHEME_WHITELIST. ([@jeremyevans](https://github.com/jeremyevans))
+- Remove `rack.multithread`/`rack.multiprocess`/`rack.run_once`. These variables generally come too late to be useful. ([#1720](https://github.com/rack/rack/pull/1720), [@ioquatix], [@jeremyevans]))
+- Remove deprecated `Rack::Request::SCHEME_WHITELIST`. ([@jeremyevans])
 
 ### Added
 
-- `Rack::Headers` added to support lower-case header keys. ([@jeremyevans](https://github.com/jeremyevans))
-- `Rack::Utils#set_cookie_header` now supports `escape_key: false` to avoid key escaping.  ([@jeremyevans](https://github.com/jeremyevans))
+- `Rack::Headers` added to support lower-case header keys. ([@jeremyevans], [@ioquatix])
```
Comment (jeremyevans) [inline CHANGELOG.md:29]: I believe only people who wrote the related code for the CHANGELOG entry should be listed.  While you did propose the idea of enforcing lower case response header keys, as far as I am aware, you did not write any code related to `Rack::Headers`, so I think you should remove yourself from this line.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,60 +6,62 @@ All notable changes to this project will be documented in this file. For info on
...
 
 ### Fixed
 
-- Make Rack::MockResponse handle non-hash headers. ([#1629](https://github.com/rack/rack/issues/1629), [@jeremyevans](https://github.com/jeremyevans))
-- TempfileReaper now deletes temp files if application raises an exception. ([#1679](https://github.com/rack/rack/issues/1679), [@jeremyevans](https://github.com/jeremyevans))
-- Fix using Rack::Session::Cookie with coder: Rack::Session::Cookie::Base64::{JSON,Zip}. ([#1666](https://github.com/rack/rack/issues/1666), [@jeremyevans](https://github.com/jeremyevans))
+- Make Rack::MockResponse handle non-hash headers. ([#1629](https://github.com/rack/rack/issues/1629), [@jeremyevans])
+- TempfileReaper now deletes temp files if application raises an exception. ([#1679](https://github.com/rack/rack/issues/1679), [@jeremyevans])
+- Fix using Rack::Session::Cookie with coder: Rack::Session::Cookie::Base64::{JSON,Zip}. ([#1666](https://github.com/rack/rack/issues/1666), [@jeremyevans])
```
Comment (jeremyevans) [inline CHANGELOG.md:61]: Considering the session code was extracted, we should drop post rack 2.2 CHANGELOG entries related to it, just keeping the single existing CHANGELOG entry that session code has been extracted to `rack-session` gem.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Thanks for working on this.  Looks mostly good, see two inline comments.

File: CHANGELOG.md (hunk 2 above)
Comment (ioquatix) [reply CHANGELOG.md:61]: Should we move those entries somewhere else?

File: CHANGELOG.md (hunk 1 above)
Comment (ioquatix) [reply CHANGELOG.md:29]: Okay, fair enough. I did spend a huge amount of time on this proposal though and it was really something I cared about.

Comment (ioquatix) [reply CHANGELOG.md:29]:
```suggestion
- `Rack::Headers` added to support lower-case header keys. ([@jeremyevans])
```

**Review by jeremyevans — APPROVED** [review]

**Pull request merged**